research/minigo/dualnet_model.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2018 MLBenchmark Group. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Defines DualNet model, the architecture of the policy and value network.
+
+Both the policy and value networks share ResNet as a majority of their
+architecture. The neural network consists of a residual tower (ResNet) and two
+separate "heads" for computing the policy and value respectively.
```

Comment:
Here or in readme it will be important to explain what these different heads are, and what they each output.